src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java

```diff
@@ -79,6 +132,69 @@ public static String xorOperator() {
     return isVHDL() ? " XOR " : "^";
   }
 
+  public static String addOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " + "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+  
+  public static String subOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " - "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+
+  public static String shiftlOperator(String signal, int with, int nrOfBits, boolean arithmetic) {
+    if (nrOfBits == 0)
+      return signal;
+    return isVHDL() ? LineBuffer.format("{{1}}{{2}} & {{4}}{{3}}{{4}}", signal, splitVector(with - 1 - nrOfBits, 0), "0".repeat(nrOfBits), nrOfBits == 1 ? "'" : "\"")
+                    : LineBuffer.format("{{{1}}{{2}},{{{3}}{1'b0}}}", signal, splitVector(with - 1 - nrOfBits, 0), nrOfBits);
+  }
+  
+  public static String shiftrOperator(String signal, int with, int nrOfBits, boolean arithmetic) {
+    if (nrOfBits == 0)
+      return signal;
+    if (arithmetic) {
+      return isVHDL()
+        ? LineBuffer.format("({{1}} DOWNTO 0 => {{2}}({{1}})) & {{2}}{{3}}", with - 1, signal, splitVector(with - 1, with - nrOfBits))
+        : LineBuffer.format("{ {{{1}}{{{2}}[{{1}}-1]}},{{2}}{{3}}}", with, signal, splitVector(with - 1, with - nrOfBits));
+    } else {
+      return isVHDL()
+        ? LineBuffer.format("{{1}}{{2}}{{1}} & {{3}}{{4}", nrOfBits == 1 ? "'" : "\"", "0".repeat(nrOfBits), signal, splitVector(with - 1, with - nrOfBits))
+        : LineBuffer.format("{ {{{1}}{1'b0}},{{2}}{{3}}}", with, signal, splitVector(with - 1, with - nrOfBits));
+    }
+  }
+
+  public static String sllOperator(String signal, int with, int nrOfBits) {
+    return shiftlOperator(signal, with, nrOfBits, false);
+  }
+
+  public static String slaOperator(String signal, int with, int nrOfBits) {
+    return shiftlOperator(signal, with, nrOfBits, true);
+  }
+
+  public static String srlOperator(String signal, int with, int nrOfBits) {
+    return shiftrOperator(signal, with, nrOfBits, false);
+  }
+  
+  public static String sraOperator(String signal, int with, int nrOfBits) {
+    return shiftrOperator(signal, with, nrOfBits, true);
+  }
+
+  public static String rolOperator(String signal, int with, int nrOfBits) {
+    return isVHDL() ? LineBuffer.format("{{1}}{{2}} & {{1}}{{3}}", signal, splitVector(with - 1 - nrOfBits, 0), splitVector(with - 1, with - nrOfBits))
```

Comment:
See above my comment on shifts/rotates.

Reply:
I mark this as resolved, as I agree the style for shift and rotate may get decided in a later PR.